include/drivers/shiftx_drv.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Race Capture Firmware
+ *
+ * Copyright (C) 2016 Autosport Labs
+ *
+ * This file is part of the Race Capture firmware suite
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details. You should
+ * have received a copy of the GNU General Public License along with
+ * this code. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _SHIFTX_DRV_H_
+#define _SHIFTX_DRV_H_
+
+#include "CAN.h"
+
+CPP_GUARD_BEGIN
+
+struct shiftx_configuration {
+        uint8_t orientation_inverted;
+        uint8_t brightness;
+        uint32_t can_bus;
+        uint32_t base_address;
+        uint8_t auto_brightness_scaling;
+};
+
+/**
+ * Process an incoming CAN message
+ * @param msg the CAN message to process
+ */
+void shiftx_handle_can_rx_msg(CAN_msg *msg);
+
+/**
+ * Retreive a pointer to the current runtime configuration
+ * @return pointer to struct of the shiftx_configuration
+ */
+struct shiftx_configuration * shiftx_get_config(void);
+
+
+/**
+ * Send the configuration update message to the connected ShiftX device, using the current runtime configurationi
+ */
+bool shiftx_update_config(void);
+
+/**
+ * Set a discrete LED on the ShiftX device
+ * @param led_index the index of the led to set
+ * @param leds_to_set number of LEDs to set. 0 = set all remaining LEDs
+ * @param red red color value
+ * @param green green color value
+ * @param blue blue color value
+ * @param flash flash rate, in Hz
+ * @return true if CAN message was successfully broadcasted
+ */
+bool shiftx_set_discrete_led(uint8_t led_index, uint8_t leds_to_set, uint8_t red, uint8_t green, uint8_t blue, uint8_t flash);
+
+/**
+ * Set the display value
+ * @param digit_index index of digit to set
+ * @param ascii ascii value to set (48 = '0', 49 = '1')
+ * @return true if CAN message was successfully broadcasted
+ */
+bool shiftx_set_display(uint8_t digit_index, uint8_t ascii);
+
+/**
+ * Configures the linear graph on the ShiftX device
+ * @param rendering_style 0=left->right; 1=center; 2=right->left
```

Comment:
This should be an enum type instead of a uint8_t

Reply:
done. Thanks.